In [3]:
for count in range(4):
    print("Go Bluejays!")

Go Bluejays!
Go Bluejays!
Go Bluejays!
Go Bluejays!


Go Bluejays!
Go Bluejays!
Go Bluejays!
Go Bluejays!

In [4]:
for count in range(2):
    print("Go Bluejays!")

Go Bluejays!
Go Bluejays!


Go Bluejays!
Go Bluejays!

In [6]:
for number in range(1, 6):
    print(number, end=" ")
print()

1 2 3 4 5 


1 2 3 4 5 
Because it stops before the last one

In [7]:
for number in range(0, 21, 5):
    print(number, end=" ")
print()

for number in range(5, 0, -1):
    print(number, end=" ")
print("Liftoff!")

0 5 10 15 20 
5 4 3 2 1 Liftoff!


0 5 10 15 20 
5 4 3 2 1 Liftoff!

In [8]:
total = 0

for number in range(1, 101):
    total += number
print(f"Total: {total}")

Total: 5050


5050

In [9]:
total = 0

for number in range(1, 11):
    total += number
print(f"Total: {total}")

Total: 55


55

In [10]:
word = "Hillsboro"
for letter in word:
    print(letter, end="-")
print()

H-i-l-l-s-b-o-r-o-


H-i-l-l-s-b-o-r-o-
R-a-f-a-e-l-

In [11]:
phrase = "banana bread"
a_count = 0
for letter in phrase:
    if letter == "a":
        a_count += 1
print(f"The letter a appears {a_count} times.")

The letter a appears 4 times.


The letter a appears 4 times.
The letter a appears 2 times.

In [12]:
number = int(input("Enter a number: "))
for multiplier in range(1, 6):
    print(f"{number} x {multiplier} = {number * multiplier}")

19 x 1 = 19
19 x 2 = 38
19 x 3 = 57
19 x 4 = 76
19 x 5 = 95


19 x 1 = 19
19 x 2 = 38
19 x 3 = 57
19 x 4 = 76
19 x 5 = 95

In [13]:
number = 1
while number <= 5:
    print(number, end=" ")
    number += 1
print()

1 2 3 4 5 


1 2 3 4 5 

In [15]:
RATE = 0.08
balance = 1000.0
years = 0

while balance < 2000:
    balance = balance * (1 + RATE)
    years += 1
print(f"It takes {years} years to double.")
print(f"Final balance: ${balance:.2f}")

It takes 10 years to double.
Final balance: $2158.92


It takes 15 years to double.
Final balance: $2078.93

It takes 10 years to double.
Final balance: $2158.92

In [18]:
total = 0
number = int(input("Enter a number (0 to stop): "))
while number != 0:
    total += number
    number = int(input("Enter a number (0 to stop): "))
print(f"Total: {total}")

ValueError: invalid literal for int() with base 10: ''

It stops

In [20]:
score = int(input("Enter a score from 0 to 100: "))
while score < 0 or score > 100:
    print("Out of range. Try again.")
    score = int(input("Enter a score from 0 to 100: "))
print(f"Accepted: {score}")

Accepted: 30


Accepted: 99

Print each character of a user’s name: FOR - You know exactly how many characters are in the name.
Keep asking for a password until the user types the correct: WHILE - You don’t know how many attempts it will take.
Add up the numbers from 1 to 50: FOR - The range (1–50) is fixed and predictable.
Count the years until a town’s population reaches 10,000: WHILE - You stop only when a condition becomes true.
Read test scores until the user enters -1:	WHILE - The number of scores is unknown; you wait for a sentinel value.
Print a multiplication table from 1 to 12: FOR - The loop runs a known number of times (1–12).

In [21]:
for number in range(10, 0, -3):
    print(number, end=" ")

10 7 4 1 

10 7 4 1 
10 7 4 1 

In [23]:
for number in range(5, 5):
    print(number)
print("done")

done


done
done

In [24]:
product = 1
for number in range(1, 5):
    product *= number
print(product)

24


24
24

In [25]:
count = 0
for letter in "Mississippi":
    if letter == "s":
        count += 1
print(count)

4


4
4

In [26]:
number = 10
while number > 0:
    number -= 3
print(number)

-2


-1
-2

In [27]:
number = 1
while number < 50:
    number *= 2
print(number)

64


64
64

In [28]:
count = 5
while count < 5:
    print("inside")
    count += 1
print("count is", count)

count is 5


5
count is 5

In [29]:
total = 0
for number in range(1, 4):
    total += number
    print(total, end=" ")

1 3 6 

1 3 6 

In [1]:
for number in range(1, 10):
    print(number)

1
2
3
4
5
6
7
8
9


range(1, 10) stops at 9, so 10 never prints

In [ ]:
for number in range(1, 11):
    print(number)

In [ ]:
for number in range(1, 6):
    total = 0
    total += number
print(total)

total = 0 is inside the loop, so it resets every iteration. You only add the last number

In [ ]:
total = 0
for number in range(1, 6):
    total += number
print(total)

In [ ]:
count = 5
while count > 0:
    print(count)
    print("Liftoff!")

count never changes, so the loop runs forever and prints “Liftoff!” every time

In [ ]:
count = 5
while count > 0:
    print(count)
    count -= 1
print("Liftoff!")

In [ ]:
for number in range(10, 0):
    print(number)

range(10, 0) counts up, but 10 → 0 is backwards. Python sees no valid forward step, so the loop never runs

In [ ]:
for number in range(10, 0, -1):
    print(number)

In [ ]:
number = input("Enter a number (0 to stop): ")
while number != 0:
    print(f"You entered {number}")
    number = input("Enter a number (0 to stop): ")

input() returns a string, so "0" != 0 is always true. The loop never stops

In [ ]:
number = int(input("Enter a number (0 to stop): "))
while number != 0:
    print(f"You entered {number}")
    number = int(input("Enter a number (0 to stop): "))

In [ ]:
total = 0
for count in range(3):
    score = int(input("Score: "))
    total += score
    average = total / 3
    print(f"Average: {average:.2f}")

Average is printed inside the loop, so it prints three times

In [ ]:
total = 0
for count in range(3):
    score = int(input("Score: "))
    total += score
average = total / 3
print(f"Average: {average:.2f}")

In [ ]:
n = int(input("Enter a positive integer: "))
total = 0

for number in range(2, n + 1, 2):
    total += number

print(f"The sum of even numbers from 2 to {n} is {total}.")

In [ ]:
n = int(input("Enter a non-negative integer: "))
factorial = 1

for number in range(1, n + 1):
    factorial *= number

print(f"{n}! = {factorial}")

In [ ]:
phrase = input("Enter a phrase: ")
letter = input("Enter a single letter: ")

count = 0
for ch in phrase:
    if ch == letter:
        count += 1

print(f"'{letter}' appears {count} times.")

In [ ]:
for c in range(0, 101, 10):
    f = c * 9 / 5 + 32
    print(f"{c} C = {f:.1f} F")

In [2]:
FINAL = -1

score = int(input("Enter a score (-1 to stop): "))
count = 0
total = 0

if score == FINAL:
    print("No scores entered.")
else:
    while score != FINAL:
        total += score
        count += 1
        score = int(input("Enter a score (-1 to stop): "))

    average = total / count
    print(f"You entered {count} scores. Average: {average:.2f}")

No scores entered.


In [3]:
goal = float(input("Savings goal: $"))
deposit = float(input("Monthly deposit: $"))

while deposit <= 0:
    print("Deposit must be greater than 0.")
    deposit = float(input("Monthly deposit: $"))

balance = 0
months = 0

while balance < goal:
    balance += deposit
    months += 1

print(f"You reach your goal in {months} months.")
print(f"Final balance: ${balance:.2f}")

You reach your goal in 20 months.
Final balance: $1000.00
